### PR TITLE
Fix CustomNode view layout after it is saved.

### PR DESCRIPTION
**Pull request by reddyashish**
### Purpose

Task: https://jira.autodesk.com/browse/DYN-5423. 

This PR is to fix the custom node layout by serializing the "Views" property when a new file is saved. This is done in the WorkspaceViewModel which will handle both the HomeWorkspaceModel and CustomNodeWorkspaceModel. 

CN saving

The previous check for CustomNodeWorkspaceModel is moved inside WorkspaceViewModel, where we will update the file path info in the custom node manager after a new custom node file is saved (related to the [PR](https://github.com/DynamoDS/Dynamo/pull/12687)).

### Declarations
Check these if you believe they are true
- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes
Fix CustomNode view layout after it is saved.

### Reviewers
@DynamoDS/dynamo

### CODE REVIEW

**Comment by QilongTang** [comment]
Excellent! Do you know when this regression begin to happen?

File: src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs [hunk 1]
```diff
@@ -2339,7 +2330,8 @@ public void ShowSaveDialogAndSaveResult(object parameter)
                     _fileDialog.InitialDirectory = fi.DirectoryName;
                     _fileDialog.FileName = fi.Name;
                 }
-                else if (vm.Model.CurrentWorkspace is CustomNodeWorkspaceModel)
+
```
Comment (reddyashish) [inline src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs:2333]: Removed the "else" check here as this condition for CustomNodeWorkspaceModel should be met irrespective of the FileName being null.

**Comment by reddyashish** [comment]
@QilongTang After this https://github.com/DynamoDS/Dynamo/pull/12687 fix by me, which was to store and retrieve the custom node file-path info after it is saved without the need to restart Dynamo.

**Comment by QilongTang** [comment]
> @QilongTang After this #12687 fix by me, which was to store and retrieve the custom node file-path info after it is saved without the need to restart Dynamo.

Gotcha, looks like this has been failing for a while. Would you add a unit test to guard it?

File: src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs [hunk 2]
```diff
@@ -2001,16 +2001,7 @@ private void InternalSaveAs(string path, SaveContext saveContext, bool isBackup
...
-                // If the current workspace is a CustomNodeWorkspaceModel, then call the save method on it.
-                if (CurrentSpaceViewModel.Model is CustomNodeWorkspaceModel)
-                {
-                    CurrentSpaceViewModel.Model.Save(path, false, Model.EngineController);
-                }
-                else
-                {
-                    CurrentSpaceViewModel.Save(path, isBackup, Model.EngineController, saveContext);
-                }
+                CurrentSpaceViewModel.Save(path, isBackup, Model.EngineController, saveContext);
```
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs:2004]: So for none custom node workspace, including saveContext does not matter and would not change the save file result?

Comment (reddyashish) [reply src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs:2004]: yeah no change in result. This save method was called with saveContext before this issue came up.

**Review by QilongTang — COMMENTED** [review]
LGTM with one question

**Review by QilongTang — APPROVED** [review]

**Pull request merged**